### PR TITLE
ADDED initial test structure,  first test

### PR DESCRIPTION
**Pull request by data-bas**
ADDED initial test structure,  first test

changed hashed_password -> password in User model

### CODE REVIEW

File: src/todo_app/routers/users.py [hunk 1]
```diff
@@ -3,19 +3,22 @@
...
 
+
 from ..sqlmodel_orm.models.user_model import (
     Users,
     UsersCreate,
     UsersPublic,
     UsersUpdate,
 )
 from ..dependencies import SessionDep
+from ..internal.encrypt import hash_password
```
Comment (MyrtheLammerse) [inline src/todo_app/routers/users.py]: Het is meer standaard om absolute imports te gebruiken in plaats van relatieve. Het zorgt ervoor dat de import altijd geldig blijft als de file verplaatst wordt en daarnaast is het ook duidelijker waar een bepaalde methode vandaan komt.

File: src/todo_app/sqlmodel_orm/models/todo_model.py [hunk 2]
```diff
@@ -3,8 +3,8 @@
     date as date_dtype,
 )  # change name date because field is also named date
 
-from ..database import engine
-from .user_model import Users
+from todo_app.sqlmodel_orm.database import engine
+from todo_app.sqlmodel_orm.models.user_model import Users
 
 
 class TodoBase(SQLModel):
```
Comment (MyrtheLammerse) [inline src/todo_app/sqlmodel_orm/models/todo_model.py:10]: I would say that the ids in the Todo class can never default to None and should be handled differently

File: src/todo_app/sqlmodel_orm/models/user_model.py [hunk 3]
```diff
@@ -23,13 +23,15 @@ class Users(UserBase, table=True):
...
+    password: str
 
 
 class UsersCreate(UserBase):
     email: str
+    password: str # tijdelijk toegevoegd voor tests
 
 
 class UsersUpdate(UserBase):
     email: str | None = None
```
Comment (MyrtheLammerse) [inline src/todo_app/sqlmodel_orm/models/user_model.py]: all of these variables (same as with todo_model) could use some input validation. For example, what does an email always look like? Same goes for id f.e.

File: src/todo_app/routers/todos.py [hunk 4]
```diff
@@ -1,9 +1,10 @@
 from fastapi import APIRouter, HTTPException, Query
 from sqlmodel import select
+from typing import Annotated
 
-from ..sqlmodel_orm.models.todo_model import Todo, TodoCreate, TodoRead, TodoUpdate
-from ..dependencies import SessionDep
-from ..sqlmodel_orm.models.user_model import Users
+from todo_app.sqlmodel_orm.models.todo_model import Todo, TodoCreate, TodoRead, TodoUpdate
+from todo_app.dependencies import SessionDep
```
Comment (MyrtheLammerse) [inline src/todo_app/routers/todos.py:6]: there is another relative import in dependencies.py

Comment (MyrtheLammerse) [reply src/todo_app/routers/todos.py:6]: but I can't comment on that since it is not part of this PR haha

File: src/todo_app/routers/todos.py [hunk 5]
```diff
@@ -24,9 +25,11 @@ def create_todo(todo: TodoCreate, session: SessionDep) -> Todo:
 @router.get("/", response_model=list[TodoRead])
 def read_todos(
     session: SessionDep,
-    user_id: int | None = Query(None),
-    offset: int = 0,
-    limit: int = Query(100, le=100),
+    user_id: int | None = None,
+    offset: Annotated[
+        int, Query(max_length=2, gt=0)
+    ] = 0,  # Via Annotated en Query maak je extra validatie mogelijk over de parameters
```
Comment (MyrtheLammerse) [inline src/todo_app/routers/todos.py:29]: make sure offset cannot be <0. These things would be excellent tests :)

File: src/todo_app/routers/todos.py [hunk 6]
```diff
@@ -24,9 +25,11 @@ def create_todo(todo: TodoCreate, session: SessionDep) -> Todo:
...
+    limit: Annotated[int, Query(le=100)] = 100,
 ):
     query = select(Todo)
     if user_id is not None:
```
Comment (MyrtheLammerse) [inline src/todo_app/routers/todos.py:33]: you have some duplicated code with raising the HTTPException. I would create a private method that checks if the user_id is valid

**Review by MyrtheLammerse — APPROVED** [review]

File: src/todo_app/routers/users.py (hunk 1 above)
Comment (data-bas) [reply src/todo_app/routers/users.py]: Done

File: src/todo_app/sqlmodel_orm/models/todo_model.py (hunk 2 above)
Comment (data-bas) [reply src/todo_app/sqlmodel_orm/models/todo_model.py:10]: Will implement when adding alembic

**Pull request merged**